smart-contracts/smart-contract/tests/main-contract_test.ts

```diff
@@ -48,6 +51,134 @@ const err_not_proposed_notifier = '(err u124)';
+    }
+
+    block = chain.mineBlock([
+      Tx.contractCall(CONTRACT_NAME, ADD_PENDING_MINERS, [], deployer.address),
+      Tx.contractCall(CONTRACT_NAME, GET_MINERS_LIST, [], deployer.address),
+    ]);
+    assertEquals(block.receipts[0].result, `(ok true)`);
+    assertEquals(block.receipts[1].result, `[${deployer.address}, ${waiting_list}]`);
+    assertEquals(block.receipts.length, 2);
+    assertEquals(block.height, 653);
+
+    block = chain.mineBlock([
+      Tx.contractCall(CONTRACT_NAME, ASK_TO_JOIN, [types.ascii(accounts.get(`wallet_${300}`)!.address)], accounts.get(`wallet_${300}`)!.address),
+      Tx.contractCall(CONTRACT_NAME, GET_DATA_WAITING_MINER, [types.list([types.principal(accounts.get(`wallet_${300}`)!.address)])], deployer.address),
+    ]);
+
+    for (let i = 91; i <= 299; i++) {
+      const miner = accounts.get(`wallet_${300}`)!;
+      block = chain.mineBlock([
+        Tx.contractCall(CONTRACT_NAME, VOTE_POSITIVE_JOIN, [types.principal(miner.address)], accounts.get(`wallet_${i}`)!.address),
+      ]);
+    }
+
+    for (let i = 1; i <= 90; i++) {
+      const miner = accounts.get(`wallet_${300}`)!;
+      block = chain.mineBlock([
+        Tx.contractCall(CONTRACT_NAME, VOTE_NEGATIVE_JOIN, [types.principal(miner.address)], accounts.get(`wallet_${i}`)!.address),
+      ]);
+    }
+
+    block = chain.mineBlock([
+      Tx.contractCall(CONTRACT_NAME, GET_DATA_WAITING_MINER, [types.list([types.principal(accounts.get(`wallet_${300}`)!.address)])], deployer.address),
+    ]);
+  },
+});
+
+Clarinet.test({
+  name: 'Get All Data Removals 300 Miners',
+  async fn(chain: Chain, accounts: Map<string, Account>) {
+    const deployer = accounts.get('deployer')!;
+    let waiting_list = [];
+    let block = chain.mineBlock([]);
+    assertEquals(block.receipts.length, 0);
+    assertEquals(block.height, 2);
+
+    for (let i = 1; i <= 299; i++) {
+      const miner = accounts.get(`wallet_${i}`)!;
+      block = chain.mineBlock([
+        Tx.contractCall(CONTRACT_NAME, ASK_TO_JOIN, [types.ascii(miner.address)], miner.address),
+      ]);
+      if (i == 1) waiting_list.push(`${miner.address}`);
+      else waiting_list.push(` ${miner.address}`);
+    }
+    block = chain.mineBlock([Tx.contractCall(CONTRACT_NAME, GET_WAITING_LIST, [], deployer.address)]);
+    assertEquals(block.receipts[0].result, `[${waiting_list}]`);
+    assertEquals(block.receipts.length, 1);
+    assertEquals(block.height, 302);
+
+    for (let i = 1; i <= 299; i++) {
+      const miner = accounts.get(`wallet_${i}`)!;
+      block = chain.mineBlock([
+        Tx.contractCall(CONTRACT_NAME, VOTE_POSITIVE_JOIN, [types.principal(miner.address)], deployer.address),
+        Tx.contractCall(CONTRACT_NAME, TRY_ENTER_POOL, [], miner.address),
+      ]);
+    }
+
+    block = chain.mineBlock([Tx.contractCall(CONTRACT_NAME, GET_PENDING_LIST, [], deployer.address)]);
+    assertEquals(block.receipts[0].result, `[${waiting_list}]`);
```

Comment:
<s> this is waiting_list variable but the call is pending_list </s>
rename variable on declaration to testing_list